var/spack/repos/builtin/packages/spark/package.py

```diff
@@ -0,0 +1,36 @@
+from spack import *
+
+import distutils
+from distutils import dir_util
+
+
+class Spark(Package):
+    """
+    Apache Spark is a fast and general engine
+    for large-scale data processing.
+    """
+
+    homepage = "http://spark.apache.org"
+    url      = "http://spark.apache.org/downloads.html"
+
+    variant('hadoop', default=False,
+            description='Build with Hadoop')
+
+    depends_on('jdk')
+    depends_on('hadoop', when='+hadoop')
+
+    version('2.0.0', '8a5307d973da6949a385aefb6ff747bb',
+            url = 'http://mirrors.ocf.berkeley.edu/apache/spark/spark-2.0.0/spark-2.0.0-bin-without-hadoop.tgz')
```

Comment:
same as above, `url` is used outside, no `url` in versions, unless Spack can't parse them automatically.